### PR TITLE
Test the proxy in release mode in Docker in CI on the master branch.

### PR DESCRIPTION
**Pull request by briansmith**
Previously we were not running the proxy tests in the release configuration.

Run the proxy tests in the release configuration through Docker.

Docker builds with tests in release mode are too slow to run on every
pull request so release mode tests will only be run on the master
branch.

Signed-off-by: Brian Smith <brian@briansmith.org>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -115,9 +115,9 @@ jobs:
         - export BUILD_DEBUG=1 DOCKER_TRACE=1
 
       script:
-        # Proxy tests are run in the `test` stage, re-running them here would be
-        # redundant/slow. #280
-        - PROXY_SKIP_TESTS=1 bin/docker-build
+        # We re-run the tests here in *release* (--release) mode since the
+        # `test` stage only runs them in *debug* node.
```
Comment (siggy) [inline .travis.yml]: `*debug* mode`

File: bin/_docker.sh [hunk 2]
```diff
@@ -52,6 +53,22 @@ docker_build() {
     echo "$repo:$tag"
 }
 
+docker_build_go() {
```
Comment (olix0r) [inline bin/_docker.sh]: this all seems unrelated to this change?

Comment (briansmith) [reply bin/_docker.sh]: > this all seems unrelated to this change?

Yes, bad branch management on my part. I removed that commit.
